### PR TITLE
Update sql_deadlock.md

### PR DESCRIPTION
**Pull request by pierreln-dd**
### What does this PR do? What is the motivation?

Adds a bit of extra clarity to the SQL Server deadlock guide concerning the need to having dbm setup

Motivation is from a customer question here: https://dd.slack.com/archives/C027TGQFUMR/p1733389297042209

### Merge instructions

Merge readiness:
- [ ] Ready for merge

Merge queue is enabled in this repo. To have it automatically merged after it receives the required reviews, create the PR (from a branch that follows the `<yourname>/description` naming convention) and then add the following PR comment:

```
/merge
```

### CODE REVIEW

**Review by nenadnoveljic — APPROVED** [review]

File: content/en/database_monitoring/guide/sql_deadlock.md [hunk 1]
```diff
@@ -49,7 +50,7 @@ Supported Agent versions
   GO
 ```
 
-2. In the Datadog Agent, enable deadlocks in `sqlserver.d/conf.yaml`.
+2. In addition to the [DBM for SQL Server setup][1]. In the Datadog Agent, enable deadlocks in `sqlserver.d/conf.yaml`.
```
Comment (neko-dd) [inline content/en/database_monitoring/guide/sql_deadlock.md:53]: I would make this clarification in a different place. It belongs in a prerequisites section, not inline in the procedure.

I suggest:

* Get rid of this line change
* Get rid of line 18
* Under line 20 (`## Before you begin`), add a line that says something like `You must configure Database Monitoring for your [SQL Server][1] before following the steps in this guide.`

I've added these as suggested edits.

File: content/en/database_monitoring/guide/sql_deadlock.md [hunk 2]
```diff
@@ -14,6 +14,7 @@ further_reading:
 
 The Deadlock view enables you to explore deadlock events in your SQL Server database.
 A deadlock occurs when two or more processes are unable to proceed because each is waiting for the other to release resources.
+
 This guide assumes that Database Monitoring has been configured for your [SQL Server][1].
 
 ## Before you begin
```
Comment (neko-dd) [inline content/en/database_monitoring/guide/sql_deadlock.md:20]:
```suggestion

## Before you begin

You must configure Database Monitoring for your [SQL Server][1] before following the steps in this guide.

```

File: content/en/database_monitoring/guide/sql_deadlock.md (hunk 1 above)
Comment (neko-dd) [inline content/en/database_monitoring/guide/sql_deadlock.md:53]:
```suggestion
2. In the Datadog Agent, enable deadlocks in `sqlserver.d/conf.yaml`.
```

**Comment by pierreln-dd** [comment]
Hi @neko-dd , thanks, makes sense to me. I definitely think the main issue was the DBM setup requirement line was within a paragraph that's easily skipped, so this helps.

**Review by neko-dd — APPROVED** [review]

**Pull request merged**